src/openrct2-ui/windows/Scenery.cpp

```diff
@@ -397,6 +398,22 @@ void WindowSceneryInit()
     }
 }
 
+static int32_t WindowSceneryGetWidth()
+{
+    // count scenery groups to determine window width
+    auto numTabs = 0;
+    for (ObjectEntryIndex scenerySetIndex = 0; scenerySetIndex < MAX_TABS; scenerySetIndex++)
+    {
+        const auto* sceneryGroupEntry = get_scenery_group_entry(scenerySetIndex);
+        if (sceneryGroupEntry != nullptr && scenery_group_is_invented(scenerySetIndex))
+        {
+            numTabs++;
+        }
+    }
+    // calculate initial window width
+    return std::max<int32_t>(WINDOW_SCENERY_MIN_WIDTH, 5 + numTabs * SCENERY_TAB_WIDTH);
```

Comment:
This calculation is off by one tab if there are miscellaneous items. You will have to check if that tab is present as well, and add 1 to numTabs if it is.